services/go_nn/src/internal/servers/grpcServer.go

```diff
@@ -2,55 +2,149 @@ package servers
 
 import (
 	"context"
-	//"fmt"
-	"log"
-	"net"
-
+	"fmt"
 	"github.com/PerfectStepCoder/yp_go_nn/src/internal/engine"
 	pb "github.com/PerfectStepCoder/yp_go_nn/src/internal/proto/gen"
 	"google.golang.org/grpc"
-	//"github.com/PerfectStepCoder/yp_go_nn/src/internal/storage"
-	//"google.golang.org/grpc/metadata"
-	//"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"log"
+	"net"
+	"runtime"
+	"sync"
 )
 
 // ServerGRPC поддерживает все необходимые методы сервера.
 type ServerGRPC struct {
 	nn *engine.OnnxNeuralNetwork
 	pb.UnimplementedClassifyNNServer
-	server *grpc.Server 
+	server       *grpc.Server
+	countWorkers int
 }
 
 // CreateTask - отправляем изображение для классификации.
 func (s *ServerGRPC) CreateOneTask(ctx context.Context, in *pb.TaskOneRequest) (*pb.TaskOneResponse, error) {
 	var response pb.TaskOneResponse
 
-	images, err := engine.BytesToFloat32Matrix(in.Image, int(in.Height), int(in.Width))
+	image, err := engine.BytesToFloat32Matrix(in.Image, int(in.Height), int(in.Width))
 
 	if err != nil {
 		return nil, err
 	}
 
-	labelClassNames, err := s.nn.Detect(images)
+	labelClassNames, err := s.nn.Detect(image)
 
 	response.TaskUID = in.TaskUID
 	response.ClassName = labelClassNames[0]
 
 	return &response, nil
 }
 
+// CreateBatchTask - отправляем изображение для классификации.
+func (s *ServerGRPC) CreateBatchTask(ctx context.Context, in *pb.TaskBatchRequest) (*pb.TaskBatchResponse, error) {
+	var response pb.TaskBatchResponse
+
+	images := make([][]float32, len(in.Images))
+
+	for i, image := range in.Images {
+		img, err := engine.BytesToFloat32Slice(image)
+		if err != nil {
+			fmt.Println(err)
+		}
+		images[i] = img
+	}
+
+	labelClassNames, err := s.nn.Detect(images)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	response.TaskUID = in.TaskUID
+	response.ClassNames = labelClassNames
+
+	return &response, nil
+}
+
+// CreateBatchCodeTask - отправляем батч изображений для классификации.
+func (s *ServerGRPC) CreateBatchCodeTask(ctx context.Context, in *pb.TaskBatchRequest) (*pb.TaskBatchCodeResponse, error) {
+	var response pb.TaskBatchCodeResponse
+
+	images := make([][]float32, len(in.Images))
+
+	for i, image := range in.Images {
+		img, err := engine.BytesToFloat32Slice(image)
+		if err != nil {
+			fmt.Println(err)
+		}
+		images[i] = img
+	}
+
+	labelClassCodes, err := s.nn.DetectCode(images)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	response.TaskUID = in.TaskUID
+	response.ClassCodes = engine.IntToInt32Slice(labelClassCodes)
+
+	return &response, nil
+}
+
+func (s *ServerGRPC) CreateBatchsTask(ctx context.Context, in *pb.TaskBatchsRequest) (*pb.TaskBatchsResponse, error) {
```

Comment:
Вообще названия методов очень похожи друг на друга и сложно разобраться чем они отличаются и за что отвечают. Стоит подумать над более информативные неймингом.
```suggestion
func (s *ServerGRPC) CreateBatchesTask(ctx context.Context, in *pb.TaskBatchsRequest) (*pb.TaskBatchsResponse, error) {
```